ducc/lib/podman_store.go

```diff
@@ -0,0 +1,534 @@
+			err = writeDataToCvmfs(CVMFSRepo, linkPath, []byte(lid))
+			if err != nil {
+				LogE(err).Error("Error in writing link id to podman store")
+				return err
+			}
+
+			//Create link dir
+			symlinkPath := filepath.Join(rootPath, rootfsDir, "l", lid)
+			targetPath := filepath.Join(rootPath, rootfsDir, layerdir, "diff")
+
+			err = CreateSymlinkIntoCVMFS(CVMFSRepo, symlinkPath, targetPath)
+			if err != nil {
+				LogE(err).Error("Error in creating the symlink for the Link dir")
+				return err
+			}
+		} else {
+			data, err := ioutil.ReadFile(linkPath)
+			if err != nil {
+				LogE(err).Error("Error in reading link file")
+				return err
+			}
+			layerIdMap[layer.Digest] = string(data)
+		}
+	}
+	return nil
+}
+
+// Create the lower files for diff dirs to be used by podman.
+func (img *Image) CreateLowerFiles(CVMFSRepo string, digestMap, layerIdMap map[string]string) (err error) {
+	Log().WithFields(log.Fields{"action": "Creating Lower files for diff dirs in podman store"}).Info(img.GetSimpleName())
+	manifest, err := img.GetManifest()
+	if err != nil {
+		LogE(err).Warn("Error in getting the image manifest")
+		return err
+	}
+
+	lastLowerData := ""
+	lastDigest := ""
+	for _, layer := range manifest.Layers {
+		if lastDigest != "" {
+			layerdir := digestMap[layer.Digest]
+			lowerPath := filepath.Join(rootPath, rootfsDir, layerdir, "lower")
+			if _, err := os.Stat(lowerPath); os.IsNotExist(err) {
+				lowerdata := layerIdMap[lastDigest]
+				if lastLowerData != "" {
+					lowerdata = lowerdata + ":" + lastLowerData
+				}
+				err = writeDataToCvmfs(CVMFSRepo, lowerPath, []byte(lowerdata))
+				if err != nil {
+					LogE(err).Warn("Error in writing lower files")
+					return err
+				}
+				lastLowerData = lowerdata
+			} else {
+				data, err := ioutil.ReadFile(lowerPath)
+				if err != nil {
+					LogE(err).Error("Error in reading lower file")
+					return err
+				}
+				lastLowerData = string(data)
+			}
+		}
+		lastDigest = layer.Digest
+	}
+	return nil
+}
+
+// Ingest the image config file in podman store.
+func (img *Image) IngestConfigFile(CVMFSRepo string) (err error) {
+	Log().WithFields(log.Fields{"action": "Creating config file for the image in podman store"}).Info(img.GetSimpleName())
+	manifest, err := img.GetManifest()
+	if err != nil {
+		LogE(err).Warn("Error in getting the image manifest")
+		return err
+	}
+	//generate config file path.
+	fname, err := generateConfigFileName(manifest.Config.Digest)
+	if err != nil {
+		LogE(err).Warning("Error in generating config file name")
+		return err
+	}
+
+	imageID := strings.Split(manifest.Config.Digest, ":")[1]
+	configFilePath := filepath.Join(rootPath, imageMetadataDir, imageID, fname)
+	if _, err := os.Stat(configFilePath); os.IsNotExist(err) {
+		user := img.User
+		pass, err := GetPassword()
+		if err != nil {
+			LogE(err).Warning("Unable to get the credential for downloading the configuration blog, trying anonymously")
```

Comment:
blog --> blob (or rather `snippet` as I guess it is not binary data.  Of course it's still under the "blob" url, so unsure which one's better)

Reply:
I think, for now we can go with blob.